doc/dash/smart-switch-ip-address-assigment/smart-switch-ip-address-assignment.md

```diff
@@ -0,0 +1,372 @@
+  - Support of the new `MID_PLANE_BRIDGE` table with the following information
+    - Midplane bridge configuration with the bridge IP address and list of the DPU PCIe interfaces that should be added to the bridge for the Smart Switch case.
+- sonic-cfggen utility shall be extended to generate the following sample configuration based on the `t1-smartswitch` topology:
+  - Midplane bridge and DHCP server configuration based on the DPU information provided in the HWSKU file.
+- interface-config.service shall be extended to create a midplane bridge with the corresponding configuration in the Linux Kernel.
+- DHCP server container should be included into the switch image. DHCP server feature should be enabled by default
+
+The midplane bridge configuration and DHCP server configuration shall be generated by sonic-cfggen utility. No user involvement is required.
+
+### Device data and HWSKU ##
+
+The Smart Switch platform by default shall use the `t1-smartswitch`. The topology shall be used together with the DPU information for the sample configuration generation.
+
+The HWSKU configuration shall be extended to provide the information about the DPU available in the system.
+
+**Smart Switch**
+
+```json
+{
+    "DPUS": {
+        "Dpu0" {
+            "interface":  "Dpu0",
+        },
+        "Dpu1" {
+            "interface":  "Dpu1",
+        }
+    }
+}
+```
+
+### DPU and PCIe interfaces naming convention ###
+
+To follow the SONiC interfaces naming convention names of the DPUs in the DB and the PCIe interfaces that are connected to the DPUs shall start from the "Dpu" prefix. The indexes shall start from "0" and shall represent the id of the DPU. For example Dpu0, Dpu1, etc. It is up to vendors' implementation to ensure that interfaces are assigned with the correct name during the system initialization.
+
+### Configuration generation for IP address assignment ###
+
+The configuration generation flow can be triggered in the following cases:
+
+- During the first boot of the system, when there is no Config DB configuration file available yet.
+- During configuration recovery, when the Config DB configuration file is removed and config-setup.service restart is triggered by the user.
+
+![image](./smart-switch-ip-assignment-configuration-generation.png)
+
+#### DPU IP address allocation ####
+
+The IP address allocation for the DPUs shall be the following:
+
+- Midplane bridge network plus DPU ID plus 1. For example for the bridge with the 169.254.200.254/24 IP address DPU0 IP address shall be 169.254.200.1.
+
+#### Smart Switch configuration ####
+
+Based on the preset `t1-smartswitch` default topology the configuration generated by the sonic-cfggen utility will be the following:
+
+```json
+{
+    "DEVICE_METADATA": {
+        "localhost": {
+            "switch_type": "switch",
+            "type": "SmartSwitch",
```

Comment:
This must be subtype. type is used as "LeafRouter" etc.

Reply:
done